spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-configuration-watcher-it/pom.xml

```diff
@@ -49,33 +49,48 @@
 			<version>${wiremock.version}</version>
 			<scope>test</scope>
 		</dependency>
+
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-webflux</artifactId>
+		</dependency>
+
 		<dependency>
 			<groupId>io.kubernetes</groupId>
 			<artifactId>client-java</artifactId>
-			<scope>test</scope>
 			<exclusions>
 				<exclusion>
 					<groupId>org.slf4j</groupId>
 					<artifactId>slf4j-api</artifactId>
 				</exclusion>
 			</exclusions>
+			<scope>test</scope>
 		</dependency>
 		<dependency>
 			<groupId>io.kubernetes</groupId>
 			<artifactId>client-java-extended</artifactId>
-			<scope>test</scope>
 		</dependency>
-
 		<dependency>
-			<groupId>org.testcontainers</groupId>
-			<artifactId>k3s</artifactId>
+			<groupId>com.github.docker-java</groupId>
```

Comment:
Can you elaborate on whats happening here?

Reply:
the failure reported initially was that a container could not be started, because port 9092 was already in use. 

We should not see that, because we start a container and re-use it in all integration tests, via two things:

- `withReuse(true)` when creating it
- `echo "testcontainers.reuse.enable=true" > ~/.testcontainers.properties`

So the question I had, why was an integration test trying to create a new container, when one was already present?

The answer is that testcontainers creates a hash from a running container and tries to re-use that on in every subsequent usage. The issue we faced, is because I removed `com.github.docker-java` (that is used when computing that hash), the hash computed was different, so no re-use. To make it simpler:

- `testA` _has_ `com.github.docker-java` dependency, creates hash = 123
- `testB` _does not_ have the `com.github.docker-java` dependency, creates hash = 456
- `testC` _has_ the `com.github.docker-java` dependency, creates hash = 123

My fix here was to simply revert the dependencies, but the more general fix and far more stable is to clean-up every integration test, which I will try to do.

Reply:
Meaning you want to make changes so every integration test is not required to have the dependency?

Reply:
yes, that is exactly what I was thinking of.

Reply:
Wouldn't just placing the dependency in the integration tests parent Pom do that?

Reply:
we don't need it, at all. we can remove it from all integration tests and parent pom. As a matter of fact, we can get rid from other dependencies too. I already have a PR [here](https://github.com/spring-cloud/spring-cloud-kubernetes/pull/1312) where I started this work.

Reply:
OK so by removing it from all tests then the hash would be the same and we would reuse the container?

Reply:
Great thanks for the explanation